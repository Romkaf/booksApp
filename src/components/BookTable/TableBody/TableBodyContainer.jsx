import React from 'react';
import Spinner from '@components/Spinner';
import TableBody from './TableBody';
import PropTypes from 'prop-types';

const TableBodyContainer = ({ books, loading }) => {
	if (loading) {
		return (
			<tbody>
				<tr>
					<td>
						<Spinner />
					</td>
				</tr>
			</tbody>
		);
	}

	return <TableBody books={books} />;
};

TableBody.propTypes = {
	books: PropTypes.arrayOf(PropTypes.object),
	loading: PropTypes.bool,
};

export default TableBodyContainer;
