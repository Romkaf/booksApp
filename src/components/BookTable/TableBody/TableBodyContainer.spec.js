import React from 'react';
import TableBodyContainer from './TableBodyContainer';
import { shallow } from 'enzyme';

describe('TableBodyContainer', () => {
	let component;
	const setUp = (loading) =>
		shallow(<TableBodyContainer book={[]} loading={loading} />);

	it('should render Spinner', () => {
		component = setUp(true);
		expect(component).toMatchSnapshot();
	});

	it('should render TableBody', () => {
		component = setUp(false);
		expect(component).toMatchSnapshot();
	});
});
