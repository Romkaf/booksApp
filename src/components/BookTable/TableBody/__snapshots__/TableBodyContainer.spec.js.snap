// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyContainer should render Spinner 1`] = `<Spinner />`;

exports[`TableBodyContainer should render TableBody 1`] = `<TableBody />`;
